master/internal/db/cache.go

```diff
@@ -0,0 +1,196 @@
+package db
+
+import (
+	"context"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/determined-ai/determined/master/pkg/archive"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+type modelDefFolder struct {
+	fileTree   []*experimentv1.FileNode
+	cachedTime time.Time
+	lock       sync.RWMutex
+	path       string
+}
+
+// FileCache is metadata for files cached at file system.
+type FileCache struct {
+	rootDir string
+	maxAge  time.Duration
+	caches  map[int]*modelDefFolder
+	lock    sync.Mutex
+}
+
+// NewFileCache initialize FileCache obj.
+func NewFileCache(rootDir string, maxAge time.Duration) *FileCache {
+	err := os.RemoveAll(rootDir)
+	if err != nil {
+		log.WithError(err).Errorf("failed to initialize model def cache at %s", rootDir)
```

Comment:
fix this error string; you didn't fail to initialize the model def cache, you just failed to clear the old contents of the model def cache.

Reply:
This has been fixed